tests/cnf/core/network/sriov/tests/metricsExporter.go

```diff
@@ -0,0 +1,310 @@
+				sriovmetricsdaemonset, err = daemonset.Pull(
+					APIClient, "sriov-network-metrics-exporter", NetConfig.SriovOperatorNamespace)
+
+				return err == nil
+			}, 2*time.Minute, 2*time.Second).Should(BeTrue(), "Daemonset sriov-network-metrics-exporter is not created")
+			Expect(sriovmetricsdaemonset.IsReady(2*time.Minute)).Should(BeTrue(),
+				"Daemonset sriov-network-metrics-exporter is not ready")
+
+			By("Enable Prometheus scraping for the new Sriov Metrics Exporter by labeling operator namespace")
+			sriovNs, err := namespace.Pull(APIClient, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to fetch Sriov Namespace")
+			_, err = sriovNs.WithMultipleLabels(netparam.ClusterMonitoringNSLabel).Update()
+			Expect(err).ToNot(HaveOccurred(), "Failed to update Sriov Namespace")
+		})
+
+		AfterEach(func() {
+
+			By("Removing SR-IOV configuration")
+			err := sriovenv.RemoveSriovConfigurationAndWaitForSriovAndMCPStable()
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove SR-IOV configration")
+
+			By("Cleaning test namespace")
+			err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+				netparam.DefaultTimeout, pod.GetGVR())
+			Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+		})
+
+		AfterAll(func() {
+			By("Disable Sriov Metrics Exporter feature in default SriovOperatorConfig CR")
+			defaultOperatorConfig, err = sriov.PullOperatorConfig(APIClient, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to fetch default Sriov Operator Config")
+			defaultOperatorConfig.Definition.Spec.FeatureGates = map[string]bool{"metricsExporter": false}
+			_, err = defaultOperatorConfig.Update()
+			Expect(err).ToNot(HaveOccurred(), "Failed to disable metrics in default Sriov Operator Config")
+			Eventually(func() bool { return sriovmetricsdaemonset.Exists() }, 1*time.Minute, 1*time.Second).Should(BeFalse(),
+				"sriov-metrics-exporter is not deleted yet")
+
+		})
+
+		DescribeTable("netdevice to netdevice test traffic: ",
+			func(cPfindex, sPfindex, cWorkerindex, sWorkerindex int) {
+
+				By("Define and Create SriovNodePolicy, SriovNetwork and Pod Resources")
+				var clientResources = defineTestResources("client",
+					sriovInterfacesUnderTest[cPfindex], sriovDeviceID, "netdevice",
+					workerNodeList[cWorkerindex].Object.Name, 0)
+				var serverResources = defineTestResources("server",
+					sriovInterfacesUnderTest[sPfindex], sriovDeviceID, "netdevice",
+					workerNodeList[sWorkerindex].Object.Name, 1)
+
+				cPod, _ := createTestResources(clientResources, serverResources)
+
+				By("ICMP check between client and server pods")
+				Eventually(func() error {
+					return cmd.ICMPConnectivityCheck(cPod, []string{tsparams.ServerIPv4IPAddress}, "net1")
+				}, 1*time.Minute, 2*time.Second).Should(Not(HaveOccurred()), "ICMP Failed")
+
+				By("Wait 1min for prometheus scraping")
+				time.Sleep(1 * time.Minute)
```

Comment:
Can we replace this sleep with the Eventually func that trying to get exepcted metrics during the min,

Reply:
Request is anyway going to fail before 30sec(min). So i think it is rather optimum to wait.

Reply:
We try to keep our code free of "sleep" statements and only use them if there are no other options.

Reply:
done. replaced with Eventually